passer.py

```diff
@@ -3764,6 +3764,7 @@ def processpacket(p):
 
 	#Only write out if changes have been made (if no changes have been made, no point in writing the dictionary out).  To test this, see if there are any entries in NewSuspiciousIPs.
 	if NewSuspiciousIPs:
+		SuspiciousIPs_at_end = []
```

Comment:
`SuspiciousIPs_at_end` is set on line 3771 but that's within an `if` condition that isn't always fulfilled. `SuspiciousIPs_at_end` is then used on lines 3774 and 3778. This would sometimes cause a runtime error if the condition hadn't been met to declare the `SuspiciousIPs_at_end` variable. Initializing the variable here to an empty list prevents the errors.